### PR TITLE
PIN Component / support 1-6 digits (Issue/2830)

### PR DESCRIPTION
**Pull request by henrijoss**
### Pull Request Checklist

- [x] I have read and accept the [Contributor License Agreement](https://opensource.porsche.com/docs/cla)

#### References

- Preview: https://designsystem.porsche.com/issue/2830

#### Scope

Short description of the pull request scope

#### Resolved Issue

Resolves [#2830](https://github.com/porsche-design-system/porsche-design-system/issues/2830)

### CODE REVIEW

File: packages/components-js/tests/e2e/puppeteer/specs/pin-code.e2e.ts [hunk 1]
```diff
@@ -87,6 +88,15 @@ describe('label', () => {
   });
 });
 
+describe('should render correct amount of inputs', () => {
+  it.each(PIN_CODE_LENGTHS)('length=%d', async (length) => {
```
Comment (denyo) [inline packages/components-js/tests/e2e/puppeteer/specs/pin-code.e2e.ts:92]: Should probably switch this a bit around wording wise to match the other tests.
Bildschirmfoto 2023-10-30 um 12 10 58

File: packages/components-js/src/pages/pin-code.html [hunk 2]
```diff
@@ -6,6 +6,10 @@
   <p-pin-code label="Some Label" description="Some Description"></p-pin-code>
 </div>
 
+<div class="playground light" title="should render with label, description and length=1">
+  <p-pin-code label="Some Label" description="Some Description" length="1"></p-pin-code>
```
Comment (denyo) [inline packages/components-js/src/pages/pin-code.html:10]: Do we need visual coverage for the other lenghts, too? Seems like some are missing now.

File: packages/components/CHANGELOG.md [hunk 3]
```diff
@@ -16,6 +16,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 #### Changed
 
+- `Pin Code` supports 1-6 digits ([#2859](https://github.com/porsche-design-system/porsche-design-system/pull/2859))
```
Comment (denyo) [inline packages/components/CHANGELOG.md:19]: Maybe rephrase that x, y, z values are now supported for `length` prop as we usually do.

File: packages/components/src/components/pin-code/pin-code-utils.ts [hunk 4]
```diff
@@ -90,13 +90,14 @@ export const syncHiddenInput = (
   required: boolean
 ): void => {
   setAttributes(hiddenInput, {
-    name,
+    ...(name && { name }),
```
Comment (denyo) [inline packages/components/src/components/pin-code/pin-code-utils.ts:93]: Did this produce an attribute of undefined or something else? 
If yes, then unit tests are probably incomplete or wrong?

File: packages/components/CHANGELOG.md (hunk 3 above)
Comment (henrijoss) [reply packages/components/CHANGELOG.md]: Maybe we could add a section in our README for some common changelog entry formats.

**Review by denyo — APPROVED** [review]
